### PR TITLE
Outlet integration

### PR DESCRIPTION
**Pull request by SebaKas21**

### CODE REVIEW

File: be/src/Project/Services/Inventory/Inventory.Api/Infrastructure/Entities/Outlet.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using Foundation.GenericRepository.Entities;
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace Inventory.Api.Infrastructure.Entities
+{
+    public class Outlet : Entity
+    {
+        [Required]
+        public Guid? ProductId { get; set; }
```
Comment (dawiddworak88) [inline be/src/Project/Services/Inventory/Inventory.Api/Infrastructure/Entities/Outlet.cs]: **suggestion:** nullable Guid makes no sense with required attribute. It should be:
[Required] 
public Guid ProductId { get; set; }

File: be/src/Project/Services/Inventory/Inventory.Api/IntegrationEventsHandlers/UpdatedProductIntegrationEventHandler.cs [hunk 2]
```diff
@@ -27,6 +31,7 @@ public async Task Handle(UpdatedProductIntegrationEvent @event)
         {
             if (@event.OrganisationId.HasValue)
             {
+                await this.outletService.UpdateOutletProduct(@event.ProductId, @event.ProductName, @event.ProductSku);
```
Comment (dawiddworak88) [inline be/src/Project/Services/Inventory/Inventory.Api/IntegrationEventsHandlers/UpdatedProductIntegrationEventHandler.cs]: **suggestion:** it should be in a separate Event and EventHandler

File: be/src/Project/Services/Inventory/Inventory.Api/Services/Outlets/IOutletService.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public interface IOutletService
+    {
+        Task SyncOutletAsync(SyncOutletServiceModel model);
+        Task<PagedResults<IEnumerable<SyncOutletItemServiceModel>>> GetAsync(GetOutletsServiceModel model);
+        Task DeleteAsync(DeleteOutletServiceModel model);
+        Task<Guid> CreateAsync(OutletServiceModel model);
+        Task<Guid> UpdateAsync(UpdateOutletServiceModel model);
+        Task<OutletServiceModel> GetAsync(GetOutletServiceModel model);
+        Task UpdateOutletProduct(Guid? ProductId, string ProductName, string ProductSku);
```
Comment (dawiddworak88) [inline be/src/Project/Services/Inventory/Inventory.Api/Services/Outlets/IOutletService.cs]: **suggestion:** correct English is UpdateProductOutlet

File: be/src/Project/Services/Inventory/Inventory.Api/Services/Outlets/OutletService.cs [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Inventory.Api.Services.Outlets
+{
+    public class OutletService : IOutletService
+    {
+        private readonly InventoryContext context;
+        private readonly IStringLocalizer inventortLocalizer;
```
Comment (dawiddworak88) [inline be/src/Project/Services/Inventory/Inventory.Api/Services/Outlets/OutletService.cs]: **suggestion:** typo, should be inventoryLocalizer

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]
I've only started review of outlet but you can already start working on the first remarks once you're happy with placing orders. I'll try to find more time tomorrow, so that you can work on it next week.

File: be/src/Project/Services/Inventory/Inventory.Api/IntegrationEventsHandlers/UpdatedProductIntegrationEventHandler.cs (hunk 2 above)
Comment (SebaKas21) [reply be/src/Project/Services/Inventory/Inventory.Api/IntegrationEventsHandlers/UpdatedProductIntegrationEventHandler.cs]: @dawiddworak88  Why? I understand to make separate handler but is it sens to copy the same code?

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]
1. Please merge develop branch to this branch and resolve conflicts
2. Make all the quantity fields as wide as the Quantity field and double check texts are not overlapping inputs:
image
3. Add validation with meaningful error message when none of the quanitty fields is filled up. At the moment following happens:
image
4. At the moment I'm able to add more items to basket then available (from stock or outlet). It makes no sense. It should be checked client-side and a meaniful validation message should be displayed.
5. It shouldn't be able to enter quantity less than 0 (like -5, or -10)
6. External Reference and More Info should be null in DB if not provided
image
7. Quantity should be preset to "1", except when adding from Stock (there should Stock quantity be preset to "1") or Outlet (there should Outlet quantity be set to 1)

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]
Hi @SebaKas21 ,

seems to be almost perfect. Could you please add some space (push more air) between SKU and product name? 

image

Cheers,

**Comment by dawiddworak88** [comment]
@SebaKas21:  I'd also appreciate if you could merge develop branch to this branch and resolve conflicts. I'd be thankful if you could smoke test it once again thereafter (conflicts are usually not easy to predict consequences).

**Review by dawiddworak88 — APPROVED** [review]

**Pull request merged**